lib/src/screens/wallet_list/wallet_list_page.dart

```diff
@@ -436,26 +415,50 @@ class WalletListBodyState extends State<WalletListBody> {
     await widget.authService.authenticateAction(
       context,
       onAuthSuccess: (isAuthenticatedSuccessfully) async {
-        if (!isAuthenticatedSuccessfully) {
-          return;
-        }
+        if (!isAuthenticatedSuccessfully) return;
 
         try {
-          changeProcessText(S.of(context).wallet_list_loading_wallet(wallet.name));
+          if (widget.walletListViewModel
+              .requireHardwareWalletConnection(wallet)) {
+            await Navigator.of(context).pushNamed(
+              Routes.connectDevices,
+              arguments: ConnectDevicePageParams(
+                walletType: WalletType.monero,
+                onConnectDevice: (context, ledgerVM) async {
+                  monero!.setGlobalLedgerConnection(ledgerVM.connection);
+                  Navigator.of(context).pop();
+                },
+              ),
+            );
+          }
+
+          showPopUp<void>(
+            context: context,
+            builder: (BuildContext context) => AlertWithOneAction(
+                alertTitle: S.of(context).proceed_on_device,
+                alertContent: S.of(context).proceed_on_device_description,
+                buttonText: S.of(context).cancel,
+                buttonAction: () => Navigator.of(context).pop()),
+          );
```

Comment:
wouldn't this show up in normal wallets as well?

Reply:
Nope, requireHardwareWalletConnection is only true if it is a monero hardware wallet

Reply:
yes, but this is out of the if condition scope